tfhe/src/core_crypto/algorithms/lwe_linear_algebra.rs

```diff
@@ -71,6 +72,22 @@ pub fn lwe_ciphertext_add_assign<Scalar, LhsCont, RhsCont>(
     Scalar: UnsignedInteger,
     LhsCont: ContainerMut<Element = Scalar>,
     RhsCont: Container<Element = Scalar>,
+{
+    let ciphertext_modulus = rhs.ciphertext_modulus();
+    if ciphertext_modulus.is_compatible_with_native_modulus() {
+        lwe_ciphertext_add_assign_native_mod_compatible(lhs, rhs)
+    } else {
+        lwe_ciphertext_add_assign_non_native_mod(lhs, rhs)
```

Comment:
As both non native cases do different things, I would put them in different functions